src/methods/join.js

```diff
@@ -649,7 +658,7 @@ export class JoinImplementation {
       let savedEffects;
       let savedPathConditions = [];
       if (result1 instanceof PossiblyNormalCompletion) {
-        completion = result1.value;
+        completion = result1.getNormalCompletion();
```

Comment:
It's nice that this now works, when I tried using `getNormalCompletion` in place of `value` in the past, I found they differed in a few cases.

Reply:
The changes above were needed to fix those issues.